### DISCUSSION TITLE
unclear, missing, or poor documentation

### DISCUSSION THREAD

**Discussion by dbnex14** [rabbitmq/rabbitmq-website#1573] [General]
Installed by homebrew on my Mac, however, even for the very first step when dealing with rabbitmq (admitedly, I am new to it), but it took me few hours to google for simple answers like below

1. how to start it?
2. how to stop it?  (with a terminal command, not Ctrl+C)
3. how to restart it?
4. how to check its status?
5. how to check if it is running already?
6. how to check if it has been installed?
7. how to get the currently installed version?

after installing, it says to add export to .zsch but even that is unclear, does it need to be re sourced?

I have literally spent 3 hours and still dont know how to restart or stop rabbitMQ

The documentation I was able to find in 3 hours is extremely poor or extremly high level/unclear compared to other MQ frameworks

**Comment by michaelklishin** [comment]
1. It depends on how you install it, individual guides have sections such as [Managing the service](https://www.rabbitmq.com/install-debian.html#managing-service). There's also a Docker container example where you don't have to explicitly start anything
2. `rabbitmqctl shutdown`
3. Stop it then start it or use `systemctl restart` or similar. This is very much dependent on how RabbitMQ was installed, there cannot be a single guide
4. There is a dedicated guide on [monitoring](https://www.rabbitmq.com/monitoring.html) plus one more on Prometheus
5. See number 4
6. I suspect you either know that it has been installed or you don't. `which rabbitmqctl` is usually a good enough approximation.
7. `rabbitmqctl version`, both RabbitMQ and Erlang versions are also listed in `rabbitmq-diagnostics status`

We do not get this kind of feedback, in particular the "extremely poor" part, FWIW.

**Reply by dbnex14** [reply]
Thanks and updated my question, I think it is clear now how it was installed but on Mac, you would typically use homebrew.  Regardless, that is what I did.

**Comment by michaelklishin** [comment]
Our guides do not mention `zsch`, you can search this repo and see for yourself. I suspect this is to get CLI command completion for `zsh`. This is an entirely optional step.

You haven't specified any details on how RabbitMQ was installed. How you manage nodes, restart them,
what's the most common approach to monitoring to some extent or entirely depend on that first step.

**Reply by dbnex14** [reply]
I am new to RabbitMQ so lots of stuff you are asking is what I dont know and I am also trying to find answer to.
Installed by homebrew
The installation steps I find will make it work fine until terminal / machine is restarted, then it is lost so I exported to .zshrc shell and it seams to be fine now
Eventually also found the ways to do what I was asking above as:

RabbitMQ clarifications

> After homebrew installation, add `export PATH=$PATH:/usr/local/sbin` to .zhrc file
> 
> After this resart terminal.
> 
> To check rabbitmq version rabbitmqctl version - I get 3.11.2
> 
> To check rabbitmq status rabbitmqctl status - this will print lots of information including version
> 
> To shutdown rabbitmqctl shutdown
> 
> To start rabbitmq-server

**Comment by michaelklishin** [comment]
So any possible improvements can happen with

1. Reasonably detailed information on what guides were used and how RabbitMQ was installed
2. Examples of other "messaging frameworks" with exceptional documentation to compare to
4. The a clear understanding of what the end goal is. In many cases, RabbitMQ is managed by `systemctl`/`systemd`, a container orchestration tool (from Docker tools to Kubernetes to managed Kubernetes) or the Windows service manager. So starting, stopping and checking it status would be no different from any other service

These topics can be covered in individual installation guides (in some cases, they already are as demonstrated above), in the CLI tools guide (not sure how obvious this would be) or even a new guide (that would have to cover N cases), or they would repackage e.g. Kubernetes pod and deployment management docs, which are not at all RabbitMQ-specific

**Comment by michaelklishin** [comment]
https://github.com/rabbitmq/rabbitmq-website/commit/be3d95bc51236b5a7850fe6b0c1d231291cf3d3b explains how nodes are managed in the CLI tools guide.

It also makes it easier to spot Managing Service sections in Debian, RPM and generic UNIX package guides.

I suspect similar improvements can be made to Windows installer and Kubernetes Operator guides
but in both of those cases RabbitMQ should be no different from any other service managed
by Kubernetes or the Windows Service manager.

**Reply by michaelklishin** [reply]
I have a few other improvements in mind, at the very least we can unify all the installation guides.

We can also link to MSFT docs on Windows service management and essential pod operations on Kubernetes, things like that.

**Comment by dbnex14** [comment]
Eventually figured out but still think it would be easier to have this clearly in your documentation

RabbitMQ clarifications

- After homebrew installation, add export PATH=$PATH:/usr/local/sbin to /.zhrc file
- 
- After this resart terminal.
- 
- To check rabbitmq version `rabbitmqctl version` - I get 3.11.2
- 
- To check rabbitmq status `rabbitmqctl status `- this will print lots of information including version - would be good to have some clarification what these mean
- 
- To shutdown `rabbitmqctl shutdown`
- 
- To start r`abbitmq-server`

Another confusing part is that these commands follow multiple formats, for example:

- rabbitmq version 
- rabbitmqctl shutdown
- rabbitmq-server

I am sure you have the reasons, it is just confusing to total beginners and feels inconsistent compared to for example
`rabbitmq <command> `consistant format like:

- rabbitmq version
- rabbitmq shutdown
- rabbitmq start

, but that is another point.
Thank you

**Reply by michaelklishin** [reply]
There are close to 100 commands, so they use separate tools, by design. Except for `rabbitmqctl`, they all follow the same convention:

 * `rabbitmq-plugins`
 * `rabbitmq-diagnostics`
 * `rabbitmq-queues`
 * `rabbitmq-streams`

The server startup script are entirely separate from CLI tools. It does not support any subcommands, it's always `rabbitmq-server`.

**Comment by michaelklishin** [comment]
Those who use the Homebrew formula can always run `brew info rabbitmq` and see how to start the node as a service or in the foreground.

I have updated our Homebrew guide to mention that, as well as the same operations listed in e.g. Debian and RPM installation guides.
